src/Extensions/ComposerLoaderExtension.php

```diff
@@ -69,39 +67,29 @@ public function getPackages(array $allowedTypes = null)
 
     /**
      * Provides access to the Composer repository
-     *
-     * @return RepositoryInterface
      */
-    protected function getRepository()
+    protected function getRepository(): RepositoryInterface
     {
         /** @var Composer $composer */
         $composer = $this->getComposer();
-
-        /** @var BaseRepository $repository */
-        return new CompositeRepository([
-            new ArrayRepository([$composer->getPackage()]),
-            $composer->getRepositoryManager()->getLocalRepository(),
+        return new InstalledRepository([
+            new RootPackageRepository($composer->getPackage()),
+            $composer->getRepositoryManager()->getLocalRepository()
         ]);
     }
 
     /**
      * Find all dependency constraints for the given package in the current repository and return the strictest one
-     *
-     * @param BaseRepository $repository
-     * @param string $packageName
-     * @return string
```

Comment:
Because this is going to be a new major, and I've switched to using proper method type hints

Reply:
Is the intention that anywhere we have type hints we should remove phpdoc typings? I'm generally in favour of that wherever the phpdoc doesn't provide additional context such as here - but in that case we should probably do it as a blanket action everywhere, yes?
I think we should probably leave the phpdocs alone for this PR, since making graphql4 strongly typed is already another load of work that needs to be done, and it should be done consistently (either leaving/updating them or removing them).

Reply:
> everywhere, yes?
 
No.  Most of Silverstripe was made for PHP 5, is dynamically typed and relys on docblocks for typehinting.

Where new API is created (which is what is happening here, this is for a new major after all), I'm using strong typing and not using docblock types

Reply:
I had meant "everywhere" as in "everywhere in this module where new API is provided" so I think we're on the same page there.

Also for some reason when I made that comment my brain was telling me this was on silverstripe/graphql which it definitely isn't. 😅Sorry about that. Yup I'm okay with this.